### DISCUSSION TITLE
REI Keeps crashing

### DISCUSSION THREAD

**Discussion by Viratol** [shedaniel/RoughlyEnoughItems#917] [Q & A] [answered]
this is my crash log
[crash-2022-06-13_15.24.49-client.txt](https://github.com/shedaniel/RoughlyEnoughItems/files/8893886/crash-2022-06-13_15.24.49-client.txt)

---- Minecraft Crash Report ----
// Daisy, daisy...

Time: 6/13/22, 3:24 PM
Description: Rendering entry base

java.lang.NullPointerException: Cannot invoke "java.util.List.isEmpty()" because "quads" is null
	at net.minecraft.class_918.method_23182(class_918.java:2048)
	at net.minecraft.class_918.method_23179(class_918.java:170)
	at me.shedaniel.rei.plugin.client.entry.ItemEntryDefinition$ItemEntryRenderer.renderBase(ItemEntryDefinition.java:303)
	at me.shedaniel.rei.plugin.client.entry.ItemEntryDefinition$ItemEntryRenderer.renderBase(ItemEntryDefinition.java:228)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.renderBatched(BatchedEntryRendererManager.java:177)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.render(BatchedEntryRendererManager.java:110)
	at me.shedaniel.rei.impl.client.gui.widget.entrylist.EntryListWidget.method_25394(EntryListWidget.java:315)
	at me.shedaniel.rei.impl.client.gui.ScreenOverlayImpl.renderWidgets(ScreenOverlayImpl.java:743)
	at me.shedaniel.rei.impl.client.gui.ScreenOverlayImpl.method_25394(ScreenOverlayImpl.java:642)
	at me.shedaniel.rei.RoughlyEnoughItemsCoreClient.lambda$registerEvents$20(RoughlyEnoughItemsCoreClient.java:396)
	at net.minecraft.class_757.handler$zim000$renderScreenPost(class_757.java:5057)
	at net.minecraft.class_757.method_3192(class_757.java:877)
	at net.minecraft.class_310.method_1523(class_310.java:1122)
	at net.minecraft.class_310.method_1514(class_310.java:737)
	at net.minecraft.client.main.Main.main(Main.java:236)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_918.method_23182(class_918.java:2048)
	at net.minecraft.class_918.method_23179(class_918.java:170)

-- Screen details --
Details:
	Screen name: net.minecraft.class_490
Stacktrace:
	at me.shedaniel.rei.impl.client.util.CrashReportUtils.screen(CrashReportUtils.java:50)
	at me.shedaniel.rei.impl.client.util.CrashReportUtils.essential(CrashReportUtils.java:44)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.renderBatched(BatchedEntryRendererManager.java:180)
	at me.shedaniel.rei.impl.client.gui.widget.BatchedEntryRendererManager.render(BatchedEntryRendererManager.java:110)
	at me.shedaniel.rei.impl.client.gui.widget.entrylist.EntryListWidget.method_25394(EntryListWidget.java:315)
	at me.shedaniel.rei.impl.client.gui.ScreenOverlayImpl.renderWidgets(ScreenOverlayImpl.java:743)
	at me.shedaniel.rei.impl.client.gui.ScreenOverlayImpl.method_25394(ScreenOverlayImpl.java:642)
	at me.shedaniel.rei.RoughlyEnoughItemsCoreClient.lambda$registerEvents$20(RoughlyEnoughItemsCoreClient.java:396)
	at net.minecraft.class_757.handler$zim000$renderScreenPost(class_757.java:5057)
	at net.minecraft.class_757.method_3192(class_757.java:877)
	at net.minecraft.class_310.method_1523(class_310.java:1122)
	at net.minecraft.class_310.method_1514(class_310.java:737)
	at net.minecraft.client.main.Main.main(Main.java:236)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Renderer details --
Details:
	Renderer name: me.shedaniel.rei.impl.client.gui.widget.entrylist.EntryListStackEntry
	Z level: 0
	Bounds: me.shedaniel.math.Rectangle@578a65e6[height=18,width=18,x=613,y=243]
	Notice mark: 0
	Interactable: true
	Interactable favorites: true
	Highlight enabled: true
	Tooltip enabled: true
	Background enabled: false
	Entries count: 1

-- Current Rendering Entry --
Details:
	Renderer name: me.shedaniel.rei.impl.common.entry.TypedEntryStack
	Z level: 100
	Entry type: minecraft:item
	Is empty: false
	Entry identifier: twilightforest:pink_castle_door

-- Entry Renderer --
Details:
	Entry definition class name: me.shedaniel.rei.plugin.client.entry.ItemEntryDefinition
	Item Type: pink_castle_door
	Item Damage: 0
	Item NBT: null
	Item Foil: false

-- Screen details --
Details:
	Screen name: net.minecraft.class_490

-- System Details --
Details:
	Minecraft Version: 1.18.2
	Minecraft Version ID: 1.18.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 643865360 bytes (614 MiB) / 2147483648 bytes (2048 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 4
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 3 2300X Quad-Core Processor          
	Identifier: AuthenticAMD Family 23 Model 8 Stepping 2
	Microarchitecture: unknown
	Frequency (GHz): 3.50
	Number of physical packages: 1
	Number of physical CPUs: 4
	Number of logical CPUs: 4
	Graphics card #0 name: Radeon RX 570 Series
	Graphics card #0 vendor: Advanced Micro Devices, Inc. (0x1002)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x67df
	Graphics card #0 versionInfo: DriverVersion=30.0.15021.11005
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 3.00
	Memory slot #0 type: DDR4
	Virtual memory max (MB): 23482.40
	Virtual memory used (MB): 14848.72
	Swap memory total (MB): 15360.00
	Swap memory used (MB): 714.72
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		architectury: Architectury 4.4.70
		balm-fabric: Balm 3.0.0+0
		bclib: BCLib 1.4.7
		betterend: Better End 1.1.1
		capybara: Capybara 1.0.3
		cloth-config: Cloth Config v6 6.2.57
			cloth-basic-math: cloth-basic-math 0.6.0
		detailab: Detail Armor Bar 2.6.2+1.18-fabric
		exlinefurniture: Exline's Furniture Mod 2.1.2
		fabric: Fabric API 0.55.1+1.18.2
			fabric-api-base: Fabric API Base 0.4.4+d7c144a860
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.1+2373a54560
			fabric-biome-api-v1: Fabric Biome API (v1) 8.0.4+7b0f312660
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.12+3ac43d9560
			fabric-command-api-v1: Fabric Command API (v1) 1.1.9+d7c144a860
			fabric-commands-v0: Fabric Commands (v0) 0.2.8+b4f4f6cd60
			fabric-containers-v0: Fabric Containers (v0) 0.1.22+d7c144a860
			fabric-content-registries-v0: Fabric Content Registries (v0) 3.0.6+d82b939260
			fabric-convention-tags-v1: Fabric Convention Tags 1.0.1+474e4f7960
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.1+ee261a3e60
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 3.0.0+9fcb91d260
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.19+ffb2c71e60
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.8+d7c144a860
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.19+d7c144a860
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.17+d7c144a860
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.13+d7c144a860
			fabric-item-api-v1: Fabric Item API (v1) 1.5.1+6c4e5d0960
			fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.13+91896a4960
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.12+54e5b2ec60
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.10+b4f4f6cd60
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.0.4+d8d7804a60
			fabric-loot-api-v2: Fabric Loot API (v2) 1.0.0+e747827960
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.0+e747827960
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.2+81ea5b3c60
			fabric-models-v0: Fabric Models (v0) 0.3.6+d7c144a860
			fabric-networking-api-v1: Fabric Networking API (v1) 1.0.22+e6b169eb60
			fabric-networking-v0: Fabric Networking (v0) 0.3.9+b4f4f6cd60
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 2.1.2+032c981d60
			fabric-particles-v1: Fabric Particles (v1) 0.2.12+526dc1ac60
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.9.9+0d9ab37260
			fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.13+d882b91560
			fabric-renderer-indigo: Fabric Renderer - Indigo 0.5.0+7faf0d8860
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.11+b4f4f6cd60
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.7+d7c144a860
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 2.0.2+54e5b2ec60
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.13+b4f4f6cd60
			fabric-rendering-v1: Fabric Rendering (v1) 1.10.7+54e5b2ec60
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.0.3+91b7aa6660
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.5.1+e747827960
			fabric-screen-api-v1: Fabric Screen API (v1) 1.0.10+d882b91560
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.2.1+1f6558e860
			fabric-textures-v0: Fabric Textures (v0) 1.0.12+3ac43d9560
			fabric-transfer-api-v1: Fabric Transfer API (v1) 1.6.2+f4563ac860
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 1.1.0+e747827960
		fabrichomes: FabricHomes 2.2.1
		fabricloader: Fabric Loader 0.14.7
		friendsandfoes: Friends&Foes 1.4.3
			mm: Manningham Mills 2.3
		gobber2: Gobber2 2.5.74
			pugh_lib: Pugh Lib 1.3.0
			pugh_tools: Pugh Tools 1.2.5
		imperishableitems: Imperishable Items 2.1.2
		java: OpenJDK 64-Bit Server VM 17
		lambdynlights: LambDynamicLights 2.1.0+1.17
			pride: Pride Lib 1.1.0+1.17
			spruceui: SpruceUI 3.3.0+1.17
				org_aperlambda_lambdajcommon: lambdajcommon 1.8.1
		minecraft: Minecraft 1.18.2
		mining_dims: Mining Dimensions 1.3.7
			customportalapi: Custom Portal Api 0.0.1-beta50-1.18
		mobhealthbar: MobHealthBar 1-18-1-fabric
		morevillagers-fabric: MoreVillagersFabric 3.2.0-SNAPSHOT
		owo: oωo 0.7.2-no-cme+1.18
		patchouli: Patchouli 1.18.2-69-FABRIC
			fiber: fiber 0.23.0-2
		roughlyenoughitems: Roughly Enough Items 8.2.479
		simple_backpack: SimpleBackpack 1.2.37
		sit: Sit 1.18.2-18
		sodium: Sodium 0.4.1+build.15
			org_joml_joml: joml 1.10.2
		the_aether: Paradise Lost 1.6.9+1.18.2
			crowdin-translate: CrowdinTranslate 1.4+1.18.2
			gud_recipe_confidence: Recipe Confidence 1.0.2
			incubus_core: Incubus Core 1.6.6+1.18.2
			moretags: More Tags 3.0.4
			reach-entity-attributes: Reach Entity Attributes 2.1.1
			satin: Satin 1.7.2
		thonkutil: ThonkUtil 2.14.1+1.18
			thonkutil-base: ThonkUtil Base 1.12.2+23c1dc2c60
			thonkutil-capes-v1: ThonkUtil Capes (v1) 1.3.2+9e23427060
			thonkutil-coords-v1: ThonkUtil Coords (v1) 1.0.19+9b80223760
			thonkutil-customization-v1: ThonkUtil Customization (v1) 1.0.14+9b80223760
			thonkutil-legacy: ThonkUtil Legacy 1.0.20+9b80223760
			thonkutil-modchecker-v1: ThonkUtil ModChecker (v1) 1.0.12+9b80223760
			thonkutil-potions-v0: ThonkUtil Potions (v0) 1.4.1+9b80223760
			thonkutil-titlescreen-v1: ThonkUtil TitleScreen (v1) 1.1.9+9b80223760
			thonkutil-trades-v1: ThonkUtil Trades (v1) 1.1.20+9b80223760
		trinkets: Trinkets 3.3.1
		twilightforest: The Twilight Forest 4.1.232
			cardinal-components-base: Cardinal Components API (base) 4.1.4
			cardinal-components-entity: Cardinal Components API (entities) 4.1.4
			com_electronwill_night-config_core: core 3.6.3
			com_electronwill_night-config_toml: toml 3.6.3
			forge_tags: Forge Tags 2.1
			forgeconfigapiport: Forge Config API Port 3.2.0
			porting_lib: Porting Lib 1.1.0-beta+1.18.2-dev.f9b9af6
				com_github_llamalad7_mixinextras: MixinExtras 0.0.9
		universal-graves: Universal Graves 2.0.3+1.18.2
			fabric-permissions-api-v0: fabric-permissions-api 0.1-SNAPSHOT
			hologram-api: Hologram API 0.2.1+1.18-pre5
			placeholder-api: Placeholder API 1.1.3+1.17.1
			polymer: Polymer (Core) 0.2.0-rc.2+1.18.2
				packet_tweaker: Packet Tweaker 0.3.0+1.18.2
				polymer-legacy: Polymer (Compatibility) 0.1.9+0.2.0-rc.2+1.18.2
			server_translations_api: Server Translations API 1.4.12+1.18.2
			sgui: sgui 1.0.2+1.18.2
		veinmining: Vein Mining 0.0.12-1.18.2
		voicechat: Simple Voice Chat 1.18.2-2.2.45
		waystones: Waystones 10.0.0
		wi_zoom: WI Zoom 1.3-MC1.18.2
		worldedit: WorldEdit 7.2.10+1742f98

**Comment by shedaniel** [comment, marked as answer]
Caused by twillight forest, please report the issue to them instead
